noman.el

```diff
@@ -32,33 +32,34 @@
 (require 'cl-lib)
 (require 'ansi-color)
 
-(defvar-local noman--last-command nil
-  "The last command that noman executed.")
+(defgroup noman nil "Command line help reader." :group 'applications :prefix "noman-")
 
-(defvar-local noman--buttons '()
-  "A list of buttons in the current noman buffer.")
+(defcustom noman-parsing-functions
+  '(("aws" . noman--make-aws-button))
+  "Alist of form: ((COMMAND . PARSER)).
+COMMAND is a string matched against the requested command.
+PARSER  is a function which accepts a line of text and returns a button or nil."
+  :type 'alist)
```

Comment:
@TODO: This type could be more specific. 
See https://www.gnu.org/software/emacs/manual/html_node/elisp/Customization-Types.html